### PR TITLE
support PriorityClass on frontend

### LINKED ISSUE #27: support PriorityClass
Now we don't support PriorityClass.

If you don't know about PriorityClass and preemption, see this doc: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/

We cannot simulate preemption without PriorityClass. So I'd like to add the feature to create/edit/delete PriorityClass.
The implementation may take some time because development of this feature requires changes both frontend and backend. Seeing implementation of other resources will help.

Please let me know if you have any questions.

/kind feature

### PR DESCRIPTION
**Pull request by khalilswdp**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add frontend to consume api for priorityclass from backend.
#### Which issue(s) this PR fixes:

Related #27 
Fixes #27 
#### Special notes for your reviewer:

Just a question, was I supposed to separate PRs by making two separate branches (with separate commits only relevant for PR, like here) or make two branches that might share some code and commits: for example, I leave same previous branch and create a branch on commit where work on backend ended and branch on commit where work on frontend ended and make PRs for both? What are the best practices to follow next time? 
This is the front end commits after separating them from branch previous branch (front end won't work since it requires back end changes to be present).
(Answered)

/label tide/merge-method-squash

### CODE REVIEW

**Comment by sanposhiho** [comment]
> Just a question, ....

In most cases, I think the best way is to wait to create frontend PR until backend PR is merged.
Since the specification of a new backend API is not approved until backend PR is merged, it may need to be changed even if the front-end is developed before backend PR is merged.

But, for this feature, the API for PriorityClass should be exactly the same as the API of other resources, so I think it is okay to develop web frontend before backend PR is merged.
However, as you said, this frontend changes don't work and won't be reviewed until backend PR is merged.

**Comment by sanposhiho** [comment]
@khalilswdp The PR for backend API was merged. 
If you rebase this branch with the latest master, you can check the behavior of this frontend change with backend.

**Comment by sanposhiho** [comment]
@khalilswdp 
We created the pull request description.
https://raw.githubusercontent.com/kubernetes-sigs/kube-scheduler-simulator/master/.github/PULL_REQUEST_TEMPLATE.md

Could you please update the pull request description in the format of this?

**Comment by khalilswdp** [comment]
Fixes #27

**Comment by sanposhiho** [comment]
@khalilswdp Thanks for updating! 👍  I'll review it lator.

File: web/components/PriorityClassList.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            v-for="(p, i) in priorityclasses"
+            :key="i"
+            class="ma-2"
+            color="primary"
+            outlined
+            large
+            label
+            @click.stop="onClick(p)"
+          >
+            <img src="/sc.svg" height="40" alt="p.metadata.name" class="mr-2" />
```
Comment (sanposhiho) [inline web/components/PriorityClassList.vue:17]: `sc.svg` is the image for strage classes.
I googled the icon for priority class, but it seems there is no icon for priority class. 😓
So, let's delete this ``

**Review by sanposhiho — COMMENTED** [review]
Nice contribution. 👍 
I think almost all the changes are okay.

File: web/components/PriorityClassList.vue (hunk 1 above)
Comment (khalilswdp) [reply web/components/PriorityClassList.vue]: I have just deleted it

File: web/components/lib/templates/priorityclass.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+metadata:
+  name: default-priority-class
```
Comment (sanposhiho) [inline web/components/lib/templates/priorityclass.yaml:2]: nits: 
```suggestion
  name: priority-class
```

File: web/components/lib/templates/priorityclass.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+value: 1000
+globalDefault: true
+description: "Default priority class for all pods"
```
Comment (sanposhiho) [inline web/components/lib/templates/priorityclass.yaml:5]: nits: If my English is strange, please change it to a proper English sentence. (I'm not a native speaker.)

```suggestion
description: "This is a template priority class for all pods"
```

File: web/components/ResourceAddButton.vue [hunk 4]
```diff
@@ -114,6 +124,11 @@ export default defineComponent({
           // if store.count = 0, name suffix is 1.
           targetTemplate = storageclassTemplate((store.count + 1).toString());
           break;
+        case "PriorityClass":
+          store = priorityclassstore;
+          // if store.count = 0, name suffix is 1.
+          targetTemplate = priorityclassTemplate((store.count + 1).toString());
```
Comment (sanposhiho) [inline web/components/ResourceAddButton.vue:130]: There are two Priority classes from the beginning. ("system-cluster-critical" and "system-node-critical")
> Kubernetes already ships with two PriorityClasses: system-cluster-critical and system-node-critical. These are common classes and are used to ensure that critical components are always scheduled first.
https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/

```suggestion
          // if store.count = 2, name suffix is 1.
          targetTemplate = priorityclassTemplate((store.count - 1).toString());
```

**Review by sanposhiho — COMMENTED** [review]
added small comments :)

**Comment by khalilswdp** [comment]
Hi @sanposhiho I followed the proposed suggestions

File: web/components/lib/templates/priorityclass.yaml (hunk 3 above)
Comment (khalilswdp) [reply web/components/lib/templates/priorityclass.yaml]: I am not a native english speaker myself, but it look good to me

**Review by sanposhiho — APPROVED** [review]
I approve this change. Thanks @khalilswdp! 

/lgtm 
/approve

**Pull request merged**